testing/e2e/e2e_withdrawal_test.go

```diff
@@ -0,0 +1,262 @@
+		},
+	)
+	if err != nil {
+		return "", crypto.BLSPubkey{}, fmt.Errorf("failed to get validators: %w", err)
+	}
+	if len(validatorsResp.Data) == 0 {
+		return "", crypto.BLSPubkey{}, errors.New("no validators found")
+	}
+
+	// Find a validator with execution withdrawal credentials (starting with 0x01)
+	var validatorIndex string
+	var blsPubkey crypto.BLSPubkey
+	for _, validator := range validatorsResp.Data {
+		credentials := validator.Validator.WithdrawalCredentials
+
+		// Create beacon-kit's WithdrawalCredentials type
+		var beaconKitCredentials ctypes.WithdrawalCredentials
+		copy(beaconKitCredentials[:], credentials)
+
+		// Check if it has execution withdrawal credentials
+		if beaconKitCredentials.IsValidEth1WithdrawalCredentials() {
+			validatorIndex = fmt.Sprintf("%d", validator.Index)
+			// Convert the phase0.BLSPubKey to our crypto.BLSPubkey type
+			copy(blsPubkey[:], validator.Validator.PublicKey[:])
+			return validatorIndex, blsPubkey, nil
+		}
+	}
+
+	return "", crypto.BLSPubkey{}, errors.New("no validator with execution withdrawal credentials found")
+}
+
+// TestSubmitPartialWithdrawalTransaction tests submitting a partial withdrawal transaction via the withdrawal contract
+func (s *BeaconKitE2ESuite) TestSubmitPartialWithdrawalTransaction() {
+	// Use timeout context to better control the test
+	ctx, cancel := context.WithTimeout(s.Ctx(), suite.DefaultE2ETestTimeout)
+	defer cancel()
+
+	// Initialize test client
+	client := s.initBeaconTest()
+
+	// Find a validator with execution credentials
+	validatorIndex, blsPubkey, err := s.findValidatorWithExecutionCredentials(client)
+	s.Require().NoError(err)
+	s.T().Logf("Found validator with index %s for withdrawal test", validatorIndex)
+
+	// Set withdrawal amount
+	withdrawalAmount := beaconmath.Gwei(DefaultWithdrawalAmount)
+
+	// Create an rpc client using the load balancer URL
+	rpcClient, err := rpc.Dial(s.JSONRPCBalancer().URL())
+	s.Require().NoError(err)
+	defer rpcClient.Close()
+
+	rpcWrapper := &rpcWrapper{Client: rpcClient}
+
+	// Get current block number before withdrawal
+	blkNum, err := s.JSONRPCBalancer().BlockNumber(ctx)
+	s.Require().NoError(err)
+	s.T().Logf("Block number before withdrawal: %d", blkNum)
+
+	// Check for pending partial withdrawals before submitting the transaction
+	pendingWithdrawalsBefore, err := s.getPendingPartialWithdrawals(utils.StateIDHead)
+	s.Require().NoError(err)
+	s.Require().Len(pendingWithdrawalsBefore, 0, "Expected no pending withdrawals initially")
+
+	// Get the withdrawal fee
+	fee, err := eip7002.GetWithdrawalFee(ctx, rpcWrapper)
+	s.Require().NoError(err)
+	s.T().Logf("Withdrawal fee: %s wei", fee.String())
+
+	// Create the withdrawal transaction data
+	withdrawalTxData, err := eip7002.CreateWithdrawalRequestData(blsPubkey, withdrawalAmount)
+	s.Require().NoError(err)
+
+	// Use a pre-loaded key that has funds
+	privateKey, err := ethcrypto.HexToECDSA("fffdbb37105441e14b0ee6330d855d8504ff39e705c3afa8f859ac9865f99306")
```

Comment:
would explicitly say here in comments: not just any pre-loaded key with funds, but the private key **specifically of** the [credentials wallet](https://github.com/berachain/beacon-kit/blob/main/kurtosis/src/nodes/consensus/beacond/launcher.star#L73)

Reply:
makes sense, added comments.